bottlecap/src/lifecycle/invocation/triggers/step_function_event.rs

```diff
@@ -0,0 +1,366 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use sha2::{Digest, Sha256};
+
+use crate::{
+    lifecycle::invocation::triggers::{Trigger, FUNCTION_TRIGGER_EVENT_SOURCE_TAG},
+    traces::{
+        context::{Sampling, SpanContext},
+        propagation::text_map_propagator::DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY,
+    },
+};
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct LegacyStepFunctionEvent {
+    #[serde(rename = "Payload")]
+    pub payload: StepFunctionEvent,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct StepFunctionEvent {
+    #[serde(rename = "Execution")]
+    pub execution: Execution,
+    #[serde(rename = "State")]
+    pub state: State,
+    #[serde(rename = "StateMachine")]
+    pub state_machine: Option<StateMachine>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct Execution {
+    #[serde(rename = "Id")]
+    id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct State {
+    #[serde(rename = "Name")]
+    name: String,
+    #[serde(rename = "EnteredTime")]
+    entered_time: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub struct StateMachine {
+    #[serde(rename = "Id")]
+    id: String,
+}
+
+impl Trigger for StepFunctionEvent {
+    fn new(payload: serde_json::Value) -> Option<Self>
+    where
+        Self: Sized,
+    {
+        let p = payload.get("Payload").unwrap_or(&payload);
+        match serde_json::from_value::<StepFunctionEvent>(p.clone()) {
+            Ok(event) => Some(event),
+            Err(e) => {
+                tracing::debug!("Failed to deserialize Step Function Event: {e}");
+                None
+            }
+        }
+    }
+
+    fn is_match(payload: &serde_json::Value) -> bool
+    where
+        Self: Sized,
+    {
+        // Check first if the payload is a Legacy Step Function event
+        let p = payload.get("Payload").unwrap_or(payload);
+
+        let execution_id = p
+            .get("Execution")
+            .and_then(Value::as_object)
+            .and_then(|e| e.get("Id"));
+        let state = p.get("State").and_then(Value::as_object);
+        let name = state.and_then(|s| s.get("Name"));
+        let entered_time = state.and_then(|s| s.get("EnteredTime"));
+
+        execution_id.is_some() && name.is_some() && entered_time.is_some()
+    }
+
+    fn enrich_span(&self, _span: &mut datadog_trace_protobuf::pb::Span) {}
+
+    fn get_tags(&self) -> HashMap<String, String> {
+        HashMap::from([(
+            FUNCTION_TRIGGER_EVENT_SOURCE_TAG.to_string(),
+            "states".to_string(),
+        )])
+    }
+
+    fn get_arn(&self, _region: &str) -> String {
+        if let Some(sm) = &self.state_machine {
+            return sm.id.clone();
+        }
+
+        String::new()
+    }
+
+    fn get_carrier(&self) -> HashMap<String, String> {
```

Comment:
No carrier, we generate the span context. The decision behind this is that the customer might be using W3C or other style, and we cannot just generate a carrier based on that, its easier to just return a `SpanContext`